### PR TITLE
Create Sphinx travis build for the Documentation site

### PR DESCRIPTION
**Pull request by kumuditha-udayanga**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [ ] Bug fix
- [x] Feature implementation
- [ ] Doc updates

* **Related Issue**

#48 

* **What changes have you introduced?**
- Create a script to deploy documentation to gh-pages 
- Add sphinx material theme to the documentation site.

* **Does this PR introduce a breaking change?**
No

* **Preview / Steps to verify your work**:
https://kumuditha-udayanga.github.io/pslab-documentation/

# Screenshot

Screenshot 2020-01-19 at 23 41 36

Screenshot 2020-01-19 at 23 42 09

### CODE REVIEW

**Comment by kumuditha-udayanga** [comment]
@mariobehling I would be nice to have your reviews :)

**Comment by kumuditha-udayanga** [comment]
Ok sir

**Comment by kumuditha-udayanga** [comment]
@mariobehling  The changes have been done 🙂😇 Please review :)

**Comment by kumuditha-udayanga** [comment]
Sorry still I'm working on this

**Comment by kumuditha-udayanga** [comment]
@mariobehling @CloudyPadmal  The changes have been done :) In the current .md files all image links are broken I fix them too. 🙂😇

**Comment by mariobehling** [comment]
Thanks. 

Is it possible to use md files instead of rst files? Sphinx has that option. Could you make it work with that here?

Could we have a simpler website style, please?
Instead of three parts, only use the left and middle bar, please. Show the content of the right bar on the left bar, please.

**Comment by mariobehling** [comment]
Please make the top left bar of the documentation to direct to the start page of the documentation.

Screenshot from 2020-01-19 16-03-19

**Comment by mariobehling** [comment]
Instead of having that increased height bar, could we only have a standard normal height bar? The effect is surely cool, but it does not really serve a purpose except showing one additional word.

Screenshot from 2020-01-19 16-05-10

File: docs/sources/tutorials.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
+=========
```
Comment (mariobehling) [inline docs/sources/tutorials.rst]: The best would be if the "GitHub view" would be the same as the website view meaning, the way you implement that here with rst files is different to what people see when opening GitHub and looking at the md files.

File: docs/sources/index.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (mariobehling) [inline docs/sources/index.rst]: In the past using Yaydoc we achieved to use the "Readme.md" as the source for index.html. I wonder if you could achieve that here too. So, the what we see in the repo would reflect what we see in the website start.

**Comment by kumuditha-udayanga** [comment]
@mariobehling I'm working on it  😇

**Comment by kumuditha-udayanga** [comment]
@mariobehling Changes have been done 😇

**Comment by mariobehling** [comment]
> Instead of having that increased height bar, could we only have a standard normal height bar? The effect is surely cool, but it does not really serve a purpose except showing one additional word.
> 
> Screenshot from 2020-01-19 16-05-10

Please check if you could also adjust this one.

**Comment by mariobehling** [comment]
The Gyroscope on the left bar and the page itself is not following the standard. Could you please adjust this as well to finalize the PR?

Screenshot from 2020-01-20 14-54-59

**Comment by kumuditha-udayanga** [comment]
@mariobehling fixed! 😇😇

**Comment by mariobehling** [comment]
Please do read the following. Posting it again:

> > Instead of having that increased height bar, could we only have a standard normal height bar? The effect is surely cool, but it does not really serve a purpose except showing one additional word.
> > Screenshot from 2020-01-19 16-05-10
> 
> Please check if you could also adjust this one.

Furthermore the search box disappeared. A major reason for using Sphinx is that it provides simply search capabilities. Please try to get the search bar back.

**Comment by kumuditha-udayanga** [comment]
> Please do read the following. Posting it again:
> 
> > > Instead of having that increased height bar, could we only have a standard normal height bar? The effect is surely cool, but it does not really serve a purpose except showing one additional word.
> > > Screenshot from 2020-01-19 16-05-10
> > 
> > 
> > Please check if you could also adjust this one.
> 
> Furthermore the search box disappeared. A major reason for using Sphinx is that it provides simply search capabilities. Please try to get the search bar back.

@mariobehling I'm working on it  😇

**Comment by kumuditha-udayanga** [comment]
@mariobehling @adityastic  Changes have been done 😇

**Comment by mariobehling** [comment]
So, the next step would be to make the Travis build actually work. Could you help with that?

[Build Status](https://www.travis-ci.org/fossasia/pslab-documentation)

**Comment by kumuditha-udayanga** [comment]
I'm working on it :)

**Comment by pranavkarthik10** [comment]
This looks great, doesn't it need to be connected to the domain docs.pslab.io now?

**Comment by kumuditha-udayanga** [comment]
Yes @pranavkarthik10 there was issue in CNAME, a PR has been opened now.

**Pull request merged**